ops/model.py

```diff
@@ -593,7 +593,7 @@ def add_secret(self, content: Dict[str, str], *,
             owner='unit')
         return Secret(self._backend, id=id, label=label, content=content)
 
-    def open_port(self, protocol: typing.Literal['tcp', 'udp', 'icmp'],
+    def open_port(self, protocol: typing.Literal['tcp', 'udp', 'icmp'] = 'tcp',
```

Comment:
Seems like at this point it might be good to add a couple of examples to the docstring, one with the default protocol and one explicit.

```
For example::

    self.unit.open_port('udp', 4000)  # open UDP port 4000
    self.unit.open_port(port=8080)    # open TCP port 8080 (protocol is implicit)
```

Hmm, I'm having second thoughts about this API addition. Seems like we're adding too many ways to do it, and `open_port(port=8080)` looks a bit odd with the word "port" appearing twice, and is barely shorter than `open_port('tcp', 8080)` in any case. So maybe we should guide people to the new `set_ports` API but keep this one simple and explicit. Let's discuss.

Reply:
Per voice discussion, we're going to back out this change -- doesn't add much (except adding one more way to do things!). And we'll focus people on the new, simpler `set_ports` API instead.

Reply:
Discussed and we agree that it's a lot of complication to save two characters (plus the ugly double "port"), so I've stripped that out.